dashboard/modules/runtime_env/runtime_env_agent.py

```diff
@@ -288,7 +288,11 @@ async def _setup_runtime_env(
             # TODO(chenk008): Add log about allocated_resource to
             # avoid lint error. That will be moved to cgroup plugin.
             per_job_logger.debug(f"Worker has resource :" f"{allocated_resource}")
-            context = RuntimeEnvContext(env_vars=runtime_env.env_vars())
+            context = RuntimeEnvContext(
+                executable=runtime_env.executable(),
+                args=runtime_env.args(),
+                env_vars=runtime_env.env_vars(),
+            )
```

Comment:
it seems like there is a system that's meant to handle these kinds of extra args via "plugins".  is there a reason we can't use that?  I'm wary of introducing anything here that'll make it harder to upstream things, adn adding additional special cases here when there seems to already be a system in place for runtime env configuration seems like it could be one of those things.

Reply:
I'm not sure we can use a plugin here as any data that is processed by a plugin doesn't seem to be processed...

I took a quick peak into how `java_jars` is setup as that gets passed into the `RuntimeEnvContext`. That doesn't seem to use a plugin but hooks in a little differently.

I'll call out I'd definitely upstream this change separately from the rest of our changes as I'm not sure the Ray maintainers will like this setup as it allows clients to execute arbitrary commands on workers. Maybe they won't care as you could just do something similar in your Ray task but it seems prudent to keep this separate

Reply:
> I took a quick peak into how java_jars is setup as that gets passed into the RuntimeEnvContext. That doesn't seem to use a plugin but hooks in a little differently.

The `java_jars` setting does use a plugin. See `python/ray/_private/runtime_env/java_jars.py`. I agree I should probably use a plugin but possibly that should be a follow up so we don't get held up here.

Reply:
https://github.com/beacon-biosignals/Ray.jl/issues/122